deidentification/utils/get_tags_diff.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+
+import difflib
+import pydicom
+import os, re
+import argparse
+
+def get_tag_diff(path_raw, path_ano, outpath=None):
+    """
+    Return diff.txt file in the sam directory as path_raw which
+    contain the tag difference between the 2 dcm 
+
+    Args:
+        path_dcm1 (str): path of the first dcm (ref)
+        path_dcm2 (str): path of the second dcm (current)
+        outpath (str): path of the diff_file
+    """
+    print(f"Check difference between :\n{path_raw} and\n{path_ano}\n")
+    dcm_ref = pydicom.read_file(path_raw)
+    dcm_current = pydicom.read_file(path_ano)
+    diff = difflib.ndiff(str(dcm_ref).splitlines(keepends=True),
+                         str(dcm_current).splitlines(keepends=True))
+    
+    diff_from_dcm1 = []
+    diff_from_dcm_2 = []
+    global_modif = []
+    current_line_diff = None
+    previous_tag = None
+    previous_sign =  None
+
+    for i in diff:
+        if "(" not in i or ")" not in i:
+            continue
+        tag = i[i.index('('): i.index(')')+1]
+        txt = ""
+
+        # tag present in only one dcm
+        if tag != previous_tag:
+            if i[0] == "-":
+                txt = "Dcm1 only:"
+                current_line_diff = diff_from_dcm1
+            elif i[0] == "+":
+                txt = "Dcm2 only:"
+                current_line_diff = diff_from_dcm_2
+            line = txt + i[1:]
+        # between 2 diff lines : same tag + remove from first line and add to second -> value modification
+        elif tag == previous_tag and i[0] != previous_sign:
+            line = ""
+            txt = f"Modification (dcm1 -> dcm2): {tag}"
+            previous_elem = current_line_diff.pop(-1)
+            # check tag name
+            previous_name = previous_elem[previous_elem.index(')') + 1: previous_elem.index(re.search(' [A-Z]{2}:', previous_elem).group())].replace("  ", "")
+            current_name = i[i.index(')') + 1: i.index(re.search(' [A-Z]{2}:', i).group())].replace("  ", "")
+            if current_name != previous_name:
+                line += txt + f'\tName modification\t {previous_name} ---> {current_name}\n'
+
+            # check tag value
+            previous_value = previous_elem.split(': ')[-1].replace('\n', '')
+            current_value = i.split(': ')[-1].replace('\n', '') 
+            if previous_value != current_value:
+                line += txt + "\tValue modification\t" + previous_value + " -----> " + current_value + '\n'
```

Comment:
The line is not exactly created in the same way between Name modification and Value modification, (f-string or not).
In the result file, there is one space more after `Name modification\t` than after `Value modification\t`.

Reply:
It seems easier to me to see if the difference is coming from the name or/and the value when the structure of the line are not the same. I will use the same structure since you rather like it

Reply:
If the difference was intended, it's ok, I first thought it was a mistake, that's why I mentioned it.